src/styles/text/text.js

```diff
@@ -416,6 +414,26 @@ Object.assign(TextStyle, {
         }
 
         return labels;
+    },
+
+    // Build one or more labels for a line geometry
+    buildLineLabels (size, line, options, labels) {
+        let subdiv = Math.min(options.subdiv, line.length);
```

Comment:
@karimnaaji I think the issue you found is corrected just by changing this to `Math.min(options.subdiv, line.length - 1);`. That avoids a segment with `0` for both start and end, and keeps them evenly spaced. (I keep forgetting that we need to limit to line *segments*, not *points*, hence the `- 1`.) I think it's more straightforward to keep the different label placements to their own separate segment ranges (rather than letting multiple labels try the whole line range, which can only be less efficient since multiple labels would try the same segment, and only one can win).